components/mdx/Demo.tsx

```diff
@@ -117,16 +127,29 @@ font-weight: bold;
       <div className="relative h-full overflow-hidden">
         <ErrorBoundary>
           {inViewport && data ? (
-            <SandpackRunner
-              customSetup={{
-                files: data,
-                entry: `/${JSON.parse(data['/package.json']).main}`,
-                main: `/${JSON.parse(data['/package.json']).main}`,
-              }}
-              options={{
-                showNavigator: false,
-              }}
-            />
+            showCode ? (
+              <Sandpack
+                customSetup={{
+                  files: data,
+                  entry: `/${JSON.parse(data['/package.json']).main}`,
+                  main: mainFile || `/${JSON.parse(data['/package.json']).main}`,
+                }}
+                options={{
+                  showTabs: false,
+                }}
+              />
+            ) : (
+              <SandpackRunner
+                customSetup={{
+                  files: data,
+                  entry: `/${JSON.parse(data['/package.json']).main}`,
+                  main: mainFile || `/${JSON.parse(data['/package.json']).main}`,
+                }}
+                options={{
+                  showNavigator: false,
+                }}
+              />
+            )
           ) : null}
```

Comment:
We don't have to return `null` in this first case, but I'd avoid chaining ternaries anyway.

Reply:
We do, i didn't have it at the start and if the call took longer it broke because the error boundary didn't return anything

Reply:
I actually also find it more readable if you return null since saying && doesn't always work and doing a ternary always does